mlflow/tensorflow.py

```diff
@@ -13,76 +13,213 @@
+    >>> import tensorflow as tf
+    >>> tf_graph = tf.Graph()
+    >>> tf_sess = tf.Session(graph=tf_graph)
+    >>> with tf_graph.as_default():
+    >>>     signature_definition = mlflow.tensorflow.load_model(path="model_path", tf_sess=tf_sess)
+    >>>     input_tensors = [tf_graph.get_tensor_by_name(input_signature.name)
+    >>>                      for _, input_signature in signature_def.inputs.items()]
+    >>>     output_tensors = [tf_graph.get_tensor_by_name(output_signature.name)
+    >>>                       for _, output_signature in signature_def.outputs.items()]
+    """
+    if run_id is not None:
+        path = _get_model_log_dir(model_name=path, run_id=run_id)
+    m = Model.load(os.path.join(path, 'MLmodel'))
+    if FLAVOR_NAME not in m.flavors:
+        raise Exception("Model does not have {} flavor".format(FLAVOR_NAME))
+    conf = m.flavors[FLAVOR_NAME]
+    saved_model_dir = os.path.join(path, conf['saved_model_dir'])
+    return _load_model(tf_saved_model_dir=saved_model_dir, tf_sess=tf_sess,
+                       tf_meta_graph_tags=conf['meta_graph_tags'],
+                       tf_signature_def_key=conf['signature_def_key'])
+
+
+def _load_model(tf_saved_model_dir, tf_sess, tf_meta_graph_tags, tf_signature_def_key):
+    """
+    Load a specified Tensorflow model consisting of a Tensorflow meta graph and signature definition
+    from a serialized Tensorflow `SavedModel` collection.
+
+    :param tf_saved_model_dir: The local filesystem path or run-relative artifact path to the model.
+    :param tf_sess: The Tensorflow session in which to the load the metagraph.
+    :param tf_meta_graph_tags: A list of tags identifying the model's metagraph within the
+                               serialized `SavedModel` object. For more information, see the `tags`
+                               parameter of the `tf.saved_model.builder.SavedModelBuilder` method:
+                               https://www.tensorflow.org/api_docs/python/tf/saved_model/builder/
+                               SavedModelBuilder#add_meta_graph
+    :param tf_signature_def_key: A string identifying the input/output signature associated with the
+                                 model. This is a key within the serialized `SavedModel`'s signature
+                                 definition mapping. For more information, see the
+                                 `signature_def_map` parameter of the
+                                 `tf.saved_model.builder.SavedModelBuilder` method.
+    :return: A Tensorflow signature definition of type:
+             `tensorflow.core.protobuf.meta_graph_pb2.SignatureDef`. This defines input and
+             output tensors within the specified metagraph for inference.
+    """
+    meta_graph_def = tf.saved_model.loader.load(
+            sess=tf_sess,
+            tags=tf_meta_graph_tags,
+            export_dir=tf_saved_model_dir)
+    signature_def = tf.contrib.saved_model.get_signature_def_by_key(
```

Comment:
Was thinking we should try to avoid using `tf.contrib` if we can, since it'll be deprecated / split up in TF 2.0 ([see roadmap]( https://www.tensorflow.org/community/roadmap#roadmap_2)). However the roadmap isn't super clear on the plan for `tf.contrib` & I can't find an equivalent utility method outside of contrib, so seems fine to leave this as is & refactor later

Reply:
Good point! Unfortunately, I also can't seem to find an equivalent utility. I added a TODO with a link to the roadmap as a reminder to switch away from `tf.contrib` when an alternative becomes available.